addons/barcodes/static/src/js/barcode_events.js

```diff
@@ -0,0 +1,158 @@
+                var new_event = new Event("keypress", {
+                    'bubbles': old_event.bubbles,
+                    'cancelable': old_event.cancelable,
+                });
+
+                new_event.viewArg = old_event.viewArg;
+                new_event.ctrl = old_event.ctrl;
+                new_event.alt = old_event.alt;
+                new_event.shift = old_event.shift;
+                new_event.meta = old_event.meta;
+                new_event.char = old_event.char;
+                new_event.key = old_event.key;
+                new_event.charCode = old_event.charCode;
+                new_event.keyCode = old_event.keyCode || old_event.which; // Firefox doesn't set keyCode for keypresses, only keyup/down
+                new_event.which = old_event.which;
+                new_event.dispatched_by_barcode_reader = true;
+
+                old_event.target.dispatchEvent(new_event);
+            }
+        }
+    },
+
+    element_is_editable: function(element) {
+        return $(element).is('input,textarea,[contenteditable="true"]');
+    },
+
+    // This checks that a keypress event is either ESC, TAB or an
+    // arrow key. This is Firefox specific, in Chrom{e,ium}
+    // keypress events are not fired for these types of keys, only
+    // keyup/keydown.
+    is_special_key: function(e) {
+        if (e.key === "ArrowLeft" || e.key === "ArrowRight" ||
+            e.key === "ArrowUp" || e.key === "ArrowDown" ||
+            e.key === "Escape" || e.key === "Tab") {
+            return true;
+        } else {
+            return false;
+        }
+    },
+
+    // The keydown and keyup handlers are here to disallow key
+    // repeat. When preventDefault() is called on a keydown event
+    // the keypress that normally follows is cancelled.
+    keydown_handler: function(e){
+        if (this.key_pressed[e.which]) {
+            e.preventDefault();
+        } else {
+            this.key_pressed[e.which] = true;
+        }
+    },
+
+    keyup_handler: function(e){
+        this.key_pressed[e.which] = false;
+    },
+
+    handler: function(e){
+        if (! e.dispatched_by_barcode_reader && ! this.is_special_key(e)) {
+            // We only stop events targeting elements that are editable. We
+            // do not stop events targeting other elements because we have
+            // no way of redispatching 'genuine' key events that trigger
+            // native event handlers of elements. So this means that our
+            // fake events will not appear in eg. an <input> element.
+            if (! this.element_is_editable(e.target) || e.target.getAttribute("barcode_events") === "true") {
+                this.buffered_key_events.push(e);
+                e.preventDefault();
+                e.stopImmediatePropagation();
+
+                clearTimeout(this.timeout);
+                if (String.fromCharCode(e.which).match(this.suffix) !== null)
+                    this.handle_buffered_keys();
```

Comment:
I think brackets lacks aesthetics, as a character. But I guess that's not a very strong reason not to use them